WordPress/create-block-theme
# Fix problem with zip file creation on Windows

Fixes #602

This PR fixes an issue where files are not created correctly when using certain file archivers on Windows OS.

Where the problem occurs, the file is generated by `ZipArchive::addFromString`. [The PHP documentation](https://www.php.net/manual/en/ziparchive.addfromstring.php) mentions the following:

> Note: For maximum portability, it is recommended to always use forward slashes (/) as directory separator in ZIP filenames.

I don't know the exact cause, but I think backslashes are causing problems with certain file archivers.

## Testing Instructions

- Check out this branch on Windows.
- Download & install [7-Zip.](https://www.7-zip.org/)
- Run "Export Zip" in the Site Editor.
- Unzip the file using 7-zip.

## Screenshot

| Before | After |
|--------|--------|
| ![image](https://github.com/WordPress/create-block-theme/assets/54422211/70260309-5568-427a-927e-9c454c6b8f76) | ![image](https://github.com/WordPress/create-block-theme/assets/54422211/51845984-0a8d-4939-a200-9b5d6a45ca0b) |
